rewrite-javascript/src/main/java/org/openrewrite/javascript/markers/TypeReferencePrefix.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.javascript.markers;
+
+import lombok.Value;
+import lombok.With;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.marker.Marker;
+
+import java.util.UUID;
+
+@Value
+@With
+public class TypeReferencePrefix implements Marker {
+    UUID id;
+    Space prefix;
```

Comment:
I wonder what we should do about this. I know that this is how this was done for Kotlin, but we specifically want to avoid adding any `Space` properties to markers.

I also haven't yet decided how we should address this in Python, where I for now "inherited" the solution that was implemented there, which was to wrap the return type into a `TypeHint` LST type, which then contains that additional space as its prefix. Yet another option would be to wrap the `J.MethodDeclaration`.

None of these options seem quite right to me.

Can you raise this question on the Slack channel and then once we've settled on something we can update this code. Since we really want to avoid `Space` properties in markers, I suspect we will end up changing this, however.